### PR TITLE
Add workflow_call action to generate the document

### PR DESCRIPTION
**Pull request by KeisukeShima**
Closes #8

### CODE REVIEW

**Comment by KeisukeShima** [comment]
There doesn't seem to be a way to reference the shell scripts in this repository.
Therefore, I need to write the script in the workflow file.

https://github.com/tier4/autoware-documentation-proposal/runs/4248865852?check_suite_focus=true

**Comment by kenji-miyake** [comment]
@KeisukeShima How about getting scripts by `curl`? Is it impossible?

**Comment by KeisukeShima** [comment]
@kenji-miyake Thanks!! This is nice idea! :100: 
https://github.com/tier4/autoware-documentation-proposal/runs/4249151187?check_suite_focus=true

File: .github/create-doc-comment.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+# Create comment body for generate-docs.yaml
+# Usage: create_doc_comment.sh <base_url> <base_branch>
```
Comment (kenji-miyake) [inline .github/create-doc-comment.sh:3]:
```suggestion
# Usage: create-doc-comment.sh <base_url> <base_branch>
```

File: .github/workflows/generate-docs.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        with:
+          github_token: ${{ secrets.token }}
+          publish_dir: ./public
+          keep_files: true
+
+      - name: Retrieve create-doc-comment script
+        run: |
+          curl --silent --show-error \
+            --output create-doc-comment.sh \
+            https://raw.githubusercontent.com/tier4/autoware-github-actions/feature/add-generate-docs/.github/create-doc-comment.sh
```
Comment (kenji-miyake) [inline .github/workflows/generate-docs.yaml:83]: How about merging the script first and referring to `main` branch here?

Comment (KeisukeShima) [reply .github/workflows/generate-docs.yaml]: Thanks for letting me know. I forgot to fix it.

d773426

File: .github/workflows/generate-docs.yaml [hunk 3]
```diff
@@ -0,0 +1,111 @@
+name: Generate documentation
+
+on:
+  workflow_call:
+    inputs:
+      github-pages-url:
```
Comment (kenji-miyake) [inline .github/workflows/generate-docs.yaml:6]: Can you get this from `mkdocs.yaml` like this?

```sh
grep -oP "site_url:\s+\K(.*)" mkdocs.yml
```

File: .github/create-doc-comment.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+modified_files=$(git diff --name-only "$base_branch"...HEAD)
+
+# Find modified markdown files
+markdown_paths=()
+for modified_file in $modified_files; do
+  if ! [[ $modified_file =~ \.md$ ]]; then
+    continue
+  fi
+
+  markdown_paths+=("$modified_file")
```
Comment (kenji-miyake) [inline .github/create-doc-comment.sh:42]:
```suggestion
  markdown_paths+=("${modified_file//docs\//}")
```

Comment (kenji-miyake) [reply .github/create-doc-comment.sh:42]: You can give `docs` as an argument.

File: .github/workflows/generate-docs.yaml (hunk 3 above)
Comment (kenji-miyake) [reply .github/workflows/generate-docs.yaml]: 760de30605ed29ba11e309d978436927cdd90249

**Review by kenji-miyake — APPROVED** [review]

**Review by kenji-miyake — CHANGES_REQUESTED** [review]
Waiting for the CI passes in https://github.com/tier4/autoware-documentation-proposal/pull/11

**Comment by KeisukeShima** [comment]
I'm trying to figure out how to fix it...

image

**Comment by kenji-miyake** [comment]
Probably curl is failed. You can confirm that by using `cat create-doc-command.sh`.

**Comment by KeisukeShima** [comment]
@kenji-miyake Thanks! I will check the problem after this PR merged.

**Comment by kenji-miyake** [comment]
Ah, it's because this PR isn't merged yet.
So let's merge!

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**